File: be/src/test-firebase.ts
import { initializeApp, cert, getApps } from 'firebase-admin/app';
import { getDatabase } from 'firebase-admin/database';
import * as dotenv from 'dotenv';
import * as path from 'path';
import * as fs from 'fs';

dotenv.config();

async function testFirebase() {
    try {
        console.log('🔄 Menghubungkan ke Firebase...');

        // 1. Tentukan path file kredensial JSON
        const keyPath = process.env.FIREBASE_SERVICE_ACCOUNT_PATH
            ? path.resolve(
                  process.cwd(),
                  process.env.FIREBASE_SERVICE_ACCOUNT_PATH,
              )
            : path.resolve(process.cwd(), 'serviceAccountKey.json');

        if (!fs.existsSync(keyPath)) {
            throw new Error(`File kredensial tidak ditemukan di: ${keyPath}`);
        }

        const serviceAccount = JSON.parse(fs.readFileSync(keyPath, 'utf8'));

        // 2. Inisialisasi app jika belum diinisialisasi
        const app =
            getApps().length === 0
                ? initializeApp({
                      credential: cert(serviceAccount),
                      databaseURL: process.env.FIREBASE_DATABASE_URL,
                  })
                : getApps()[0];

        const db = getDatabase(app);
        const testRef = db.ref('system_health/test_connection');

        // 3. Tulis data uji
        await testRef.set({
            status: 'CONNECTED',
            timestamp: new Date().toISOString(),
            service: 'frss-backend',
        });

        console.log('✅ Berhasil menulis data ke Firebase Realtime Database!');

        // 4. Baca kembali data
        const snapshot = await testRef.once('value');
        console.log('📥 Data terbaca dari Firebase:', snapshot.val());
    } catch (error: any) {
        console.error(
            '❌ Gagal terhubung ke Firebase:',
            error.message || error,
        );
    }
}

testFirebase();

File: be2/test_firebase.py
import os
from datetime import datetime
from app.services.firebase_service import firebase_service

def run_test():
    print("🔄 Memeriksa status inisialisasi Firebase be2...")
    if not firebase_service.is_ready:
        print("❌ Firebase service belum siap. Periksa file kredensial JSON atau .env")
        return

    ref = firebase_service.get_ref("system_health/ai_service_test")
    test_payload = {
        "status": "ONLINE",
        "timestamp": datetime.now().isoformat(),
        "service": "frss-ai-be2"
    }
    
    ref.set(test_payload)
    print("✅ Berhasil mengirim status ke Firebase Realtime Database!")
    
    data = ref.get()
    print("📥 Data yang diterima kembali:", data)

if __name__ == "__main__":
    run_test()